src/components/Endpoint.tsx

```diff
@@ -143,7 +145,7 @@ const Endpoint = (props) => {
 
         if (config.is_hub) {
           setAPIError(null);
-          setLoginURL(error_response.login_url);
+          setHubResponse(error_response);
```

Comment:
This is probably the biggest change. I'm passing the entire error response from the operation_ls failure. With this change, the hub login widget had to be refactored to handle the response.